### PR TITLE
feat: Add smart contract verification

### LINKED ISSUE #253: Add Smart Contract verification
- There should be a way for the user to start the Smart Contract Verification process

- In the first screen there should be different fields: smart contract id, compiler version (dropdown of the available versions), license type (dropdown), agree to terms of service

- In the second screen there should be a way to upload multiple files, starting from the entry file (Non-standard includes should be parsed and ask the user for additional files, recursively).

- Captcha should be enabled

- There should be a status reported the user that could be: Processing (the verification process will be handled by a queue), Fail (also with the reason), Success

### LINKED ISSUE #667: Remove workaround for adapting various error messages
### Context information

application version: 
device: 
browser: 
operating system: 

### Steps to reproduce
go to `result` page and see additional logic to adapt various types of error messaging from verifier service

### What is expected?
It's working as expected. Could be done with less frontend logic.
This will be a breaking change once https://github.com/aeternity/smart-contract-verifier/issues/39 is resolved

### What is actually happening?
There is a workaround / additional logic to adapt various error message types

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #253 
resolves #667

## Demo
This is a simple happy path on how to verify contract
All of the AEstudio sample examples should be tested @Liubov-crypto 

https://github.com/aeternity/aescan/assets/15363559/b3f87649-f0ec-48d3-8e41-5fc653a6446b

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] add verification form
- [x] ~add captcha @michele-franchi  @janmichek~
- [ ] update terms of service @michele-franchi 
- [x] test with multifile contract @janmichek 
- [x] remove workaround logic before release https://github.com/aeternity/aescan/issues/667

### CODE REVIEW

**Comment by Liubov-crypto** [comment]
For me it trows an error: 

errrr
tttt

https://github.com/aeternity/aescan/assets/69896204/7884ee76-086d-42b8-b575-9e11577a4e44

maybe it was because I was not able to create and upload a proper file with smart contract with .aes at the end. 
I tried different redactors but can't do it on my Mac :(

**Comment by janmichek** [comment]
> For me it trows an error:
> 
> errrr tttt
> 2024-02-15.5.07.31.mov
> 
> maybe it was because I was not able to create and upload a proper file with smart contract with .aes at the end. I tried different redactors but can't do it on my Mac :(

I fixed that.
I found out there is a various error message structure form verifier service.
It works, but it's kinda implementation workaround. 
I created an issue to improve it. It would be nice to process it before the release https://github.com/aeternity/aescan/issues/667

File: nuxt.config.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export default defineNuxtConfig({
       ALTERNATIVE_NETWORK_NAME: process.env.ALTERNATIVE_NETWORK_NAME,
       AE_TOKEN_ID: process.env.AE_TOKEN_ID,
       DEBUG_MODE: process.env.DEBUG_MODE,
+      CONTRACT_VERIFICATION_SERVICE_URL: process.env.CONTRACT_VERIFICATION_SERVICE_URL,
```
Comment (michele-franchi) [inline nuxt.config.ts:39]: This should be also added to .env.example, please make sure it's up to date.

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Findings:
- When I click on "Compiler version" or "License" dropdown, the UI jumps a little bit
- The UI also jumps a bit when I select a different entry file
- Upload of files and folders together does not work
  - Let's say I have the following scenario:
    - main.aes
    - lib folder (which contains a hello.aes file) 
  - I would expect to be able to drag and drop all in the "Contract Files" section and expect the following in the UI:
    - main.aes
    - lib/hello.aes
- I would keep consistency in the required messages:
  - Smart Contract ID is required (instead of: ID is required)
  - Compiler Version is required (instead of: Please select compiler)
  - License is required (instead of: Please fill license)
- The UI should occupy the entire space, there should be just a padding in the middle (var(--space-4)), for example:
image
- In case of failure I would keep the "Smart Contract Verification" title and add the error subtitle inside of the panel, like we do on other screens. For example (See 500 error page for reference):
image

**Comment by janmichek** [comment]
@michele-franchi thanks for feedback. Working on it.

I fixed the folder drop. However there is some service issue.
Reported here
https://github.com/aeternity/smart-contract-verifier/issues/44

**Comment by janmichek** [comment]
> I would keep consistency in the required messages: 

I agree it was an inconsistent and bit funky. I aligned it in a different way

File: nuxt.config.ts (hunk 1 above)
Comment (janmichek) [reply nuxt.config.ts:39]: good suggestion, updated

**Comment by Liubov-crypto** [comment]
I tried, but it doesn't work for me, probably because I can't create .aes file on my Mac: 

https://github.com/aeternity/aescan/assets/69896204/0c579a7b-54b3-400c-9ca4-750a5c86513e

@janmichek could you please suggest me some other way to verify it?

**Comment by janmichek** [comment]
> I tried, but it doesn't work for me, probably because I can't create .aes file on my Mac:
> 2024-03-06.6.11.49.mov
> 
> @janmichek could you please suggest me some other way to verify it?

1. There was a small bug with selecting entry file through direct input. I fixed that
2. please also test drag and drop functionality
3. https://www.google.com/search?client=firefox-b-d&q=create+custom+suffix+text+file+on+mac

**Comment by Liubov-crypto** [comment]
Finally I was able to do smart contracts verification. It's working for me well on testnet. Also drag and drop functionality is working as expected. LGTM

**Pull request merged**